WordPress/src/main/java/org/wordpress/android/util/DialogUtils.java

```diff
@@ -11,22 +14,55 @@
 import org.wordpress.android.widgets.OpenSansEditText;
 import org.wordpress.android.widgets.WPTextView;
 
-public class DialogUtils {
-    public static void showMyProfileDialog(Context context,
-                                           String title,
-                                           String initialText,
-                                           String hint,
-                                           boolean isMultiline,
-                                           final Callback callback) {
-        LayoutInflater layoutInflater = LayoutInflater.from(context);
+public class DialogUtils extends DialogFragment {
+
+    private static final String TITLE_TAG = "TITLE";
+    private static final String INITIAL_TEXT_TAG = "INITIAL-TEXT";
+    private static final String HINT_TAG = "HINT";
+    private static final String IS_MULTILINE_TAG = "IS-MULTILINE";
+
+    private static Context mContext;
+    private static Callback mCallback;
```

Comment:
Now, having to use a *global* static variable holding the reference to the Callback is a sign that you basically bypass the fragment lifecycle procedure. That's rarely a good idea :(

I know that the code was passing a callback as a parameter before but now it becomes more of an issue since you end up keeping a reference. There are a few ways to fix this, the most typical of which is to have the parent activity implement the callback interface and use and cast `getActivity()` in the dialog to call it. Cheers!